### PR TITLE
expose list of nodes

### PR DESCRIPTION
**Pull request by Karsten1987**
Connect to ros2/cli_tools#1

### CODE REVIEW

File: rmw/include/rmw/types.h [hunk 1]
```diff
@@ -192,6 +192,12 @@ typedef struct RMW_PUBLIC_TYPE rmw_topic_names_and_types_t
   char ** type_names;
 } rmw_topic_names_and_types_t;
 
+typedef struct RMW_PUBLIC_TYPE rmw_node_names_t
+{
+  size_t node_count;
+  char ** names;
```
Comment (dirk-thomas) [inline rmw/include/rmw/types.h]: I would either prefix both members with `node_` or none of them. In the second case the type could also be called `rmw_names_t` and potentially reused for other names in the future.

File: rmw/src/sanity_checks.c [hunk 2]
```diff
@@ -42,3 +42,22 @@ rmw_check_zero_rmw_topic_names_and_types(
   }
   return RMW_RET_OK;
 }
+
+rmw_ret_t
+rmw_check_zero_rmw_node_names(
+  rmw_node_names_t * node_names)
+{
+  if (!node_names) {
+    RMW_SET_ERROR_MSG("node_names_t is null");
```
Comment (dirk-thomas) [inline rmw/src/sanity_checks.c]: Should be: `node_names is null`

**Comment by dirk-thomas** [comment]
I would suggest connecting all other PRs to this one to group the PRs in waffle as well as have a "master" ticket to comment on with e.g. CI builds.

This also needs a PR for rmw_opensplice so that it at least continues to build / link.

File: rmw/include/rmw/types.h (hunk 1 above)
Comment (Karsten1987) [inline rmw/include/rmw/types.h]: hmm, we could even spin this further and call it `rmw_string_array_t` ?

Comment (dirk-thomas) [inline rmw/include/rmw/types.h]: Sounds good to me. At the end its only a simple `vector<string>` in C.

**Comment by Karsten1987** [comment]
any updates on this?

**Comment by dirk-thomas** [comment]
I think it would be good to have a CI job which was run for OpenSplice to ensure it compiles. Also I didn't see any test where it checks that the returned nodes names are correct. I think that would be good to cover too.

**Comment by Karsten1987** [comment]
I added a new test for ensuring that both supported rmw implementations are returning the same issue.
Can be found in here: https://github.com/ros2/rcl/blob/rosnode_list/rcl/test/rcl/test_get_node_names.cpp

build result: [Build Status](http://ci.ros2.org/job/ci_linux/2248/)

**Pull request merged**